### PR TITLE
template api related changes

### PR DESCRIPTION
**Pull request by venkateshl-wal**

### CODE REVIEW

File: src/template/template.service.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      throw new InternalServerErrorException(
+        `Failed to upload file: ${error.message}`,
+      );
+    }
+  }
+
+  async insertTemplate(version, file: Express.Multer.File): Promise<Template> {
+    if (!file) {
+      throw new InternalServerErrorException(
+        'File is required to insert a template.',
```
Comment (supriya-wal) [inline src/template/template.service.ts:43]: change error message to `Template required `

File: src/template/template.service.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      );
+    }
+    try {
+      const templateURL = await this.uploadFile(file);
+
+      const templateParam = {
+        name: file.originalname,
```
Comment (supriya-wal) [inline src/template/template.service.ts:50]: accept name from input field. if required, you can save file original name in seperate field in db.

File: src/template/template.service.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        version: version,
+        templateurl: templateURL,
+      };
+      return await this.templateModel.create(templateParam);
+    } catch (error) {
+      throw new InternalServerErrorException(
+        `Failed to insert template: ${error.message}`,
```
Comment (supriya-wal) [inline src/template/template.service.ts:57]: `Failed to upload template`

**Review by supriya-wal — CHANGES_REQUESTED** [review]

File: src/template/template.service.ts (hunk 1 above)
Comment (venkateshl-wal) [reply src/template/template.service.ts]: resolved

File: src/template/template.service.ts (hunk 2 above)
Comment (venkateshl-wal) [reply src/template/template.service.ts]: resolved

File: src/template/template.service.ts (hunk 3 above)
Comment (venkateshl-wal) [reply src/template/template.service.ts]: resolved

**Review by supriya-wal — APPROVED** [review]

**Pull request merged**